src/azure-cli/azure/cli/command_modules/cosmosdb/tests/latest/test_cosmosdb_commands.py

```diff
@@ -1410,10 +1412,77 @@ def test_cosmosdb_managed_service_identity(self, resource_group, key_vault):
             'identity_principal_id': identity_principal_id
         })
         self.cmd('az keyvault set-policy -n {kv_name} -g {rg} --object-id {identity_principal_id} --key-permissions get unwrapKey wrapKey')
-
+        
+        # System assigned identity tests
+        cmk_account = self.cmd('az cosmosdb create -n {acc} -g {rg} --locations regionName={location} failoverPriority=0 --assign-identity [system] --default-identity FirstPartyIdentity').get_output_in_json()
         cmk_account = self.cmd('az cosmosdb update -n {acc} -g {rg} --default-identity SystemAssignedIdentity').get_output_in_json()
         assert cmk_account["defaultIdentity"] == 'SystemAssignedIdentity'
 
+        id_account = self.cmd('az cosmosdb create -n {acc} -g {rg} --assign-identity [system]').get_output_in_json()
+        assert id_account["identity"]['type'] == 'SystemAssigned'
+
+        identity_output = self.cmd('az cosmosdb identity remove -n {acc} -g {rg}').get_output_in_json()
+        assert identity_output["type"] == "None"
+
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg}').get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned"
+
+        # User assigned identity tests
+        user_identity1 = self.cmd('az identity create -n {id1} -g {rg}').get_output_in_jason()
+        user_identity2 = self.cmd('az identity create -n {id2} -g {rg}').get_output_in_jason()
+
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg} --identities {0}', user_identity1["id"]).get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned,UserAssigned"
+        assert identity_output["user_identities"][0][0] == user_identity1["id"]
+        assert len(identity_output["user_identities"]) == 1
+
+        identity_output = self.cmd('az cosmosdb identity assign -n {acc} -g {rg} --identities {0}', user_identity2["id"]).get_output_in_json()
+        assert identity_output["type"] == "SystemAssigned,UserAssigned"
+        assert identity_output["user_identities"][1][0] == user_identity2["id"]
```

Comment:
Notice it's possible that the UA2 is above UA1 in the response.
So I believe you can change the logic here to check one is UA1 and another is UA2 instead of using ["user_identities"][1][0] or ["user_identities"][0][0]